web/src/reactions.js

```diff
@@ -469,44 +469,73 @@ export function set_clean_reactions(message) {
     // to make sure reify_message_id moves the data structure
     // properly.
     message.clean_reactions = new Map();
+
+    const reaction_counts_and_user_ids = [...distinct_reactions.keys()].map((local_id) => {
+        const user_ids = user_map.get(local_id);
+        return {
+            count: user_ids.length,
+            user_ids,
+        };
+    });
+    const should_display_reactors = check_should_display_reactors(reaction_counts_and_user_ids);
```

Comment:
This calculates `should_display_reactors` only once, and not in the for loop. It's moving the work to do it before creating the clean reactions instead of after.

Previously the work was `O(number of reactions)`, since each of those three steps loops over the clean reactions (or a subset of their data).

Now it happens beforehand, modifying the calculation of `reaction_counts_and_user_ids` to something that doesn't involve the message but is still `O(distinct_reactions.length)`.